src/server.cpp

```diff
@@ -919,6 +919,64 @@ void Server::AsyncRunStep(float dtime, bool initial_step)
 		}
 	}
 
+	/*
+		Send sounds if needed
+	*/
+	{
+		m_playing_sounds_time += dtime;
+
+		// We will be accessing the environment
+		MutexAutoLock lock(m_env_mutex);
+
+		ClientInterface::AutoLock clientlock(m_clients);
```

Comment:
Assuming this acquires another mutex, why is this safe from potential deadlocks?

Reply:
I have no idea if it is safe from potential deadlocks.

But I hope that it is because it is already used here:
https://github.com/minetest/minetest/blob/3bd5169aee16f73723e5cee3b624abbb83f6531e/src/server.cpp#L784-L789